### DISCUSSION TITLE
Better fixup commits in `git sl` (feature request)

### DISCUSSION THREAD

**Discussion by alan-nf** [arxanas/git-branchless#679] [Ideas]
It would be nice if `git sl` showed the first real line of fixup commits. Right now we just see the fixup line which isn't informative. If I have a stack of fixup commits `git sl` will convey very little, e.g.:

```
◯ 4ead335 8m `npm run graphql:generate"
┃
◯ 54267e0 8m chore: Upgrade TypeScript everywhere (4.4.2 => 4.9.3)
┃
◯ 259bbac 8m fixup! chore: Upgrade TypeScript everywhere (4.4.2 => 4.9.3)
┃
◯ b242e94 8m fixup! chore: Upgrade TypeScript everywhere (4.4.2 => 4.9.3)
┃
◯ dc8d6f6 8m fixup! chore: Upgrade TypeScript everywhere (4.4.2 => 4.9.3)
┃
...
```

These fixup commits all have an informative first line, e.g. what is contained in the fix. It would be nice to show this by default, if present. Something like:

```
◯ 4ead335 8m `npm run graphql:generate"
┃
◯ 54267e0 8m chore: Upgrade TypeScript everywhere (4.4.2 => 4.9.3)
┃
◯ 259bbac 8m fixup! chore: Upgrade TypeScript everywhere (4.4.2 => 4.9.3)
┃            Make VS Code use the workspace version of TypeScript
┃
◯ b242e94 8m fixup! chore: Upgrade TypeScript everywhere (4.4.2 => 4.9.3)
┃            Fix Jest tests
...
```

**Comment by arxanas** [comment]
That sounds pretty reasonable to me, if you want to try implementing it. The mechanism to add additional descriptors to commits in the smartlog is fairly straightforward, but unfortunately none of them currently produce more than one line of output.

Admittedly, I don't use fixup commits myself; I either just check out the target commit and commit/amend directly there, or I make a new commit without an explicit `fixup!` annotation and just remember where it needs to go and manually fixup it during an interactive rebase.
